File: server/src/main/java/com/example/demo/common/dto/ApiResponse.java
package com.example.demo.common.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiResponse <T> {

    private T data;

    private String message;

    private Map<String, List<String>> errors;

    // Success response
    public ApiResponse(T data, String message) {
        this.data = data;
        this.message = message;
    }

    // Error response
    public ApiResponse(Map<String, List<String>> errors) {
        this.errors = errors;
    }
}

File: server_chat/src/common/ApiResponse.js
const {isNull} = require("../utils/Utils");

const ApiResponse = ({message, data, errors}) => {
    const response = {};

    if (!isNull(message)) response.message = message;
    if (!isNull(data)) response.data = data;
    if (!isNull(errors)) response.error = errors;


    return response;
};

module.exports = ApiResponse;